spec/workers/adviser_sign_up_worker_spec.rb

```diff
@@ -0,0 +1,181 @@
+        })
+      end
+    end
+
+    describe 'sending GCSE qualification details' do
+      it "sends no for has_gcse_maths_and_english_id if they haven't got Maths or English GCSEs" do
+        application_form.maths_gcse.destroy
+        application_form.english_gcse.destroy
+
+        expect_sign_up({
+          has_gcse_maths_and_english_id: described_class::GCSE[:no],
+        })
+      end
+
+      it 'sends no for has_gcse_maths_and_english_id if they have passed Maths but not English GCSEs' do
+        application_form.english_gcse.update(grade: 'Z')
+
+        expect_sign_up({
+          has_gcse_maths_and_english_id: described_class::GCSE[:no],
+        })
+      end
+
+      it 'sends no for has_gcse_maths_and_english_id if they have passed English but not Maths GCSEs' do
+        application_form.maths_gcse.update(grade: 'Z')
+
+        expect_sign_up({
+          has_gcse_maths_and_english_id: described_class::GCSE[:no],
+        })
+      end
+
+      it "sends no for has_gcse_science_id if they haven't got a Science GCSE" do
+        application_form.science_gcse.destroy
+
+        expect_sign_up({
+          has_gcse_science_id: described_class::GCSE[:no],
+        })
+      end
+
+      it 'sends no for has_gcse_science_id if they have not passed their Science GCSE' do
+        application_form.science_gcse.update(grade: 'Z')
+
+        expect_sign_up({
+          has_gcse_science_id: described_class::GCSE[:no],
+        })
+      end
+
+      it 'sends yes for planning_to_retake_gcse_maths_and_english_id if they are completing both Maths and English GCSEs' do
+        application_form.maths_gcse.update(currently_completing_qualification: true)
+        application_form.english_gcse.update(currently_completing_qualification: true)
+
+        expect_sign_up({
+          planning_to_retake_gcse_maths_and_english_id: described_class::GCSE[:yes],
+        })
+      end
+
+      it 'sends yes for planning_to_retake_gcse_science_id if they are completing their Science GCSE' do
+        application_form.science_gcse.update(currently_completing_qualification: true)
+
+        expect_sign_up({
+          planning_to_retake_gcse_science_id: described_class::GCSE[:yes],
+        })
+      end
+    end
+  end
+
+  def expect_request_attribtues(attributes, expected_attributes)
+    expect(attributes).to include(expected_attributes)
+  end
+
+  def expect_sign_up(expected_attribute_overrides = {})
+    expect_any_instance_of(GetIntoTeachingApiClient::TeacherTrainingAdviserApi).to \
+      receive(:sign_up_teacher_training_adviser_candidate) do |_, request|
+        request_attributes = extract_attributes(request)
+        expect_request_attribtues(request_attributes, baseline_attributes.merge(expected_attribute_overrides))
+        yield(request_attributes) if block_given?
+      end
+  end
+
+  def extract_attributes(request)
+    request.to_hash.transform_keys do |key|
+      request.class.attribute_map.invert[key]
+    end
+  end
```

Comment:
What's the goal of this?

Reply:
The swagger-generated API client we use awkwardly doesn't provide a way to get a hash of the attributes in `snake_case`; it uses `camelCase` on the `to_hash` method as ultimately it uses this as the payload to the API (which is C# and accepts `camelCase`). It does, however, provide an attribute map to convert between the two:

https://github.com/DFE-Digital/get-into-teaching-api-ruby-client/blob/master/auto-generated-gem/lib/get_into_teaching_api_client/models/teacher_training_adviser_sign_up.rb#L83-L118

So this is essentially converting `camelCase` keys of the hash to `snake_case` - I'll add a comment to make that clear. I may monkey patch the API client at some point to add a convenience method for this.

Reply:
perhaps could extract a method `get_attributes_as_snake_case` or some such, as an alternative to the comment?